File: Server/routes/seller.js
const cookieParser = require("cookie-parser");
const express = require("express");
const seller = require('../models/seller');
const jwt = require("jsonwebtoken");
const Shop = express.Router();
const AuthSeller = require("../middleware/SellerSide")
const medicine = require("../models/medicine");
//SignUp method
Shop.post("/Signup",async(req,res)=>{
    try{
        const NewSeller = new seller({
            name:"Pranav",
            shopName:"MCSunay",
            address:"PICT", 
            mobile:9979731378,
            email:"j7a@gmail.com",
            licence:"N5IL",
            password:"Pranav",
            cpassword:"Pranav"
        })
        
        const SellerCreated = await NewSeller.save();
        console.log(SellerCreated);
        const token =  await SellerCreated.generateAuthToken();
        console.log(token);
        //res.cookie("jwt",token,{expires:new Date(date.now() + 80000),httpOnly:true});

    }
    catch(err){
        console.log(err);
    }
})

//Login Function
Shop.post("/login",async(req,res)=>{
    try{
        const email = req.body.email;
        const Userfound = await seller.findOne({email:email});
        const password = req.body.password;
        if(password===Userfound.password){
            const token = await Userfound.generateAuthToken();
            console.log(token);
            //res.cookie("jwt",token,{expires:new Date(date.now() + 80000),httpOnly:true});
            
            
        }
        else{
            res.send("Invalid Details")
        }

    }catch(err){
        res.status(401).send("Invalid Details");
    }
})

//for Seller to Add medicine in to the database
Shop.post("/AddMed",AuthSeller,async(req,res)=>{
    try{const token = req.header("Authorization").replace("Bearer ", "");
    const verifySeller = jwt.verify(token,process.env.SECRET_KEY);
    const SellerFound = await seller.findOne({_id:verifySeller._id});
    const NewMedicine = new  medicine({
        name:req.body.name,
        expiry:req.body.expiry,
        Quantity:req.body.Quantity,
        Shopid: SellerFound._id
    })
    const Addmedicine = await NewMedicine.save();
    console.log(Addmedicine);
    }catch(err){
       console.log(err);
    }
})

//For Seller to see his Medicines
Shop.get("/getMed/:id",AuthSeller,async(req,res)=>{
    try{
    const id = req.params.id;
    console.log(id);
    const med = await medicine.find({Shopid:id});
    res.send(med);
    }
    catch(err){
        console.log(err);
    }
})

//For Seller to Update his medicine
Shop.patch("/UpdateMed/:id",AuthSeller,async(req,res)=>{
    try{
        const id = req.params.id;
        const Updated = await medicine.findByIdAndUpdate({_id:id},req.body,{new:true});
        res.send(Updated);

    }catch(err){
        console.log(err);
    }
})

//For Seller to Delete his Medicine
Shop.delete("/DeleteMed/:id",AuthSeller,async(req,res)=>{
    try{
        const id = req.params.id;
        const Deleted = await medicine.findByIdAndDelete({_id:id});
        res.send(Deleted);
    }catch(err){
        console.log(err);
    }
})


module.exports = Shop;

File: lib/integration/seller.dart
import 'dart:convert';
import 'dart:ffi';

import "package:http/http.dart" as http;

// enter the url here
String host = "http://localhost:2700";

Future<String> signUpShopkeeper(
    String name_,
    String shopName_,
    String address_,
    int mobile_,
    String licence_,
    String email_,
    String password_,
    String cPassword_,
    String latitude_,
    String longitude_,
    String placeId_) async {
  try {
    final body = {
      "name": name_,
      "shopName": shopName_,
      "address": address_,
      "mobile": mobile_,
      "email": email_,
      "licence": licence_,
      "password": password_,
      "cpassword": cPassword_,
      "lat": latitude_,
      "long": longitude_,
      "placeId": placeId_
    };
    Map<String, String> header = {'Content-Type': "application/json"};

    var res = await http.post(Uri.parse("$host/Signup"),
        headers: header, body: jsonEncode(body));

    switch (res.statusCode) {
      case 200:
        return jsonDecode(res.body)["tokens"][0]["token"];
    }
    return res.statusCode.toString();
  } catch (e) {
    return e.toString();
  }
}

Future<String> signInShopkeeper(String email_, String password_) async {
  try {
    final body = {
      "email": email_,
      "password": password_,
    };
    Map<String, String> header = {'Content-Type': "application/json"};

    var res = await http.post(Uri.parse("$host/login"),
        headers: header, body: jsonEncode(body));

    switch (res.statusCode) {
      case 200:
        return jsonDecode(res.body)["tokens"][0]["token"];
    }
    return res.statusCode.toString();
  } catch (e) {
    print(e.toString());
    return e.toString();
  }
}

Future<String> addMedicine(String token_, String name_, String expiry_,
    int quantity_, int cost_) async {
  try {
    final header = {"Authorization": token_};
    final body = {
      "name": name_,
      "expiry": expiry_,
      "Quantity": quantity_,
      "Cost": cost_
    };

    var res = await http.post(Uri.parse("$host/AddMed"),
        headers: header, body: jsonEncode(body));

    switch (res.statusCode) {
      case 200:
        return "sucess";
    }
    return res.statusCode.toString();
  } catch (e) {
    return e.toString();
  }
}
